genai-use-cases/aws-glue-metadata-generation/how_to_generate_metadata_for_glue_data_catalog_w_bedrock.ipynb

```diff
@@ -0,0 +1,846 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "07021766-ecce-48e0-8dad-bac2a67b4b54",
+   "metadata": {},
+   "source": [
+    "## 4. Improve meta-data descriptions by adding external documentation"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7219b23c-60e7-4828-813f-a3b57457b10e",
+   "metadata": {},
+   "source": [
+    "### Ingestion workflow"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "2815cb3f-9ba0-404c-a3cc-315d9594b45a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from langchain.docstore.document import Document\n",
+    "from langchain.document_loaders import TextLoader\n",
+    "from langchain.embeddings import BedrockEmbeddings\n",
+    "from langchain.text_splitter import CharacterTextSplitter\n",
+    "from langchain.vectorstores import FAISS"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d0035f31-802d-44f3-a4e7-61d8b4ec5e73",
+   "metadata": {},
+   "source": [
+    "### Download the data from EveryPolitician.com "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "309d52d7-47f3-4fda-ab6f-ec8f34723579",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from langchain_community.document_loaders import AsyncHtmlLoader\n",
+    "# We will use an HTML Community loader to load the external documentation stored on HTLM \n",
+    "urls = [\"http://www.popoloproject.com/specs/person.html\", \"http://docs.everypolitician.org/data_structure.html\",'http://www.popoloproject.com/specs/organization.html','http://www.popoloproject.com/specs/membership.html','http://www.popoloproject.com/specs/area.html']\n",
+    "loader = AsyncHtmlLoader(urls)\n",
+    "docs = loader.load()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5b5cc1da-6886-47cd-91f1-4e07fcbc45c7",
+   "metadata": {},
+   "source": [
+    "Let's take a look at some of the content that was loaded"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3ec61e96-fe32-4130-aedf-95d6b4ccf754",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "docs[0].page_content[0:1000]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d90371f3-9fe6-4be1-a7db-768735993ada",
+   "metadata": {},
+   "source": [
+    "Next, we split the downloaded content into chunks"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "dda0cdd3-ae85-4317-a156-24208bd951d4",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "text_splitter = CharacterTextSplitter(\n",
+    "    separator='\\n',\n",
+    "    chunk_size=1000,\n",
+    "    chunk_overlap=200,\n",
+    "\n",
+    ")\n",
+    "split_docs = text_splitter.split_documents(docs)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ae6585ea-0580-48ab-bec3-71421dc1a2d0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "embedding_model = BedrockEmbeddings(\n",
```

Comment:
LangChainDeprecationWarning: The class `BedrockEmbeddings` was deprecated in LangChain 0.2.11 and will be removed in 1.0. An updated version of the class exists in the :class:`~langchain-aws package and should be used instead. To use it run `pip install -U :class:`~langchain-aws` and import as `from :class:`~langchain_aws import BedrockEmbeddings``.
  embedding_model = BedrockEmbeddings(